src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -587,28 +633,57 @@ protected FieldValueInfo completeValueForList(ExecutionContext executionContext,
             index++;
         }
 
-        CompletableFuture<List<ExecutionResult>> resultsFuture = Async.each(fieldValueInfos, FieldValueInfo::getFieldValue);
-
-        CompletableFuture<ExecutionResult> overallResult = new CompletableFuture<>();
-        completeListCtx.onDispatched(overallResult);
-
-        resultsFuture.whenComplete((results, exception) -> {
-            if (exception != null) {
-                ExecutionResult executionResult = handleNonNullException(executionContext, overallResult, exception);
-                completeListCtx.onCompleted(executionResult, exception);
-                return;
```

Comment:
This is the completeList code - before it did

CompletableFuture<List<ExecutionResult>> resultsFuture = Async.each(fieldValueInfos, FieldValueInfo::getFieldValue);

but this now gets a bit more complicated because the items in the list can be CFs or not.

Reply:
So we unwrap that Async.each and do our CF or Object code